node/src/accountant/db_access_objects/payable_dao.rs

```diff
@@ -347,48 +307,104 @@ impl PayableDaoReal {
+                Ok(wallet),
+                Ok(high_bytes),
+                Ok(low_bytes),
+                Ok(last_paid_timestamp),
+                Ok(tx_hash_opt),
+                Ok(previous_failures),
+            ) => Ok(PayableAccountWithTxInfo {
+                account: PayableAccount {
                     wallet,
                     balance_wei: checked_conversion::<i128, u128>(BigIntDivider::reconstitute(
                         high_bytes, low_bytes,
                     )),
                     last_paid_timestamp: utils::from_unix_timestamp(last_paid_timestamp),
-                    pending_payable_opt: None,
-                })
-            }
+                },
+                tx_opt: Self::maybe_construct_tx_info(tx_hash_opt, previous_failures),
+            }),
             e => panic!(
                 "Database is corrupt: PAYABLE table columns and/or types: {:?}",
                 e
             ),
         }
     }
 
+    fn maybe_construct_tx_info(
+        pending_tx_hash_opt: Option<String>,
+        previous_failures: usize,
+    ) -> Option<TxProcessingInfo> {
+        if pending_tx_hash_opt.is_some() || previous_failures > 0 {
+            Some(TxProcessingInfo {
+                pending_tx_hash_opt,
+                failures: previous_failures,
+            })
+        } else {
+            None
+        }
+    }
+
     fn stm_assembler_of_payable_cq(feeder: AssemblerFeeder) -> String {
         format!(
-            "select
-               wallet_address,
-               balance_high_b,
-               balance_low_b,
-               last_paid_timestamp
-           from
-               payable
-           {} {}
-           order by
-               {},
-               {}
-           {}",
+            "SELECT
+            p.wallet_address,
+            p.balance_high_b,
+            p.balance_low_b,
+            p.last_paid_timestamp,
+            CASE WHEN s.status LIKE '%Pending%'
+                THEN s.tx_hash
+                ELSE NULL
+            END AS pending_tx_hash,
+            /*
+               The following case stm counts up the failing attempts in the tx processing while it
+               is still ongoing.
+            */
+            CASE WHEN EXISTS (
+                    SELECT 1
+                    FROM failed_payable
+                    WHERE status NOT LIKE '%Concluded%'
+                        AND receiver_address = p.wallet_address
+                )
+                THEN (
+                    WITH nonces_of_failures_by_wallet AS
+                    (
+                        SELECT nonce
+                        FROM failed_payable
+                        WHERE receiver_address = p.wallet_address
+                    )
+                    SELECT COUNT(*)
+                    FROM nonces_of_failures_by_wallet
+                    WHERE nonce = (SELECT MAX(nonce) FROM nonces_of_failures_by_wallet)
+                )
+                ELSE 0
+            END AS recent_failures_count
```

Comment:
You can replace the entire `EXISTS` clause with this code:

```rust
(
    SELECT COUNT(*)
    FROM failed_payable
    WHERE receiver_address = p.wallet_address
        AND status NOT LIKE '%Concluded%'
        AND nonce = (
            SELECT MAX(nonce)
            FROM failed_payable
            WHERE receiver_address = p.wallet_address
                AND status NOT LIKE '%Concluded%'
        )
) AS recent_failures_count
```

It'll automatically return 0 if no cases match.